File: mall-content-interface/src/main/java/com/mall/content/service/ContentService.java
package com.mall.content.service;

import com.mall.pojo.Content;
import common.pojo.PageResult;

import java.util.List;

/**
 * 内容/广告服务接口
 * @author Wwl
 *
 */
public interface ContentService {

	/**
	 * 增加
	 * @param content
	 */
	public void add(Content content);

	/**
	 * 修改
	 * @param content
	 */
	public void update(Content content);

	/**
	 * 根据Id获取实体
	 * @param id
	 * */
	public Content findOne(Long id);

	/**
	 * 查询所有
	 * @return
	 */
	public List<Content> findAll();

	/**
	 * 查询所有-分页
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);

	/**
	 * 根据条件查询，分页
	 * @param content
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Content content, int pageNum, int pageSize);

	/**
	 * 删除
	 * @return
	 */
	public void delete(Long[] ids);


}

File: mall-manager-web/src/main/webapp/js/service/contentService.js
/**
 * 内容/广告服务层
 * Created by Wwl on 2019/3/4
 */
app.service('contentService', function($http){

    //增加
    this.add = (entity) => $http.post('../content/add.do',entity);

    //修改
    this.update = (entity) => $http.post('../content/update.do',entity);

    //删除
    this.delete = (ids) => $http.get('../content/delete.do?ids='+ids);

    //查询实体
    this.findOne = (id) => $http.get('../content/findOne.do?id='+id);

    //查询列表
    this.findList = () => $http.get('../content/list.do');

    //分页查询
    this.findPage = (page,size) => $http.get('../content/listByPage.do?page=' + page + '&size=' + size);

    //搜索
    this.search = (page,size,searchEntity) => $http.post('../content/search.do?page='+page+'&size='+size,searchEntity);

    //更新状态
    this.updateStatus = (ids,status) => $http.get('../content/updateStatus.do?ids='+ids+'&status='+status);
});